conans/client/cmake.py

```diff
@@ -74,6 +63,15 @@ def __init__(self, settings_or_conanfile, generator=None, cmake_system_name=True
         else:
             raise ConanException("First parameter of CMake() has to be a ConanFile instance.")
 
+        self._os = self._settings.get_safe("os")
+        self._compiler = self._settings.get_safe("compiler")
+        self._compiler_version = self._settings.get_safe("compiler.version")
+        self._arch = self._settings.get_safe("arch")
+        self._build_type = self._settings.get_safe("build_type")
+        self._op_system_version = self._settings.get_safe("os.version")
```

Comment:
Conan probably should also use here and "api_level" for Android, because `_cmake_cross_build_defines` use `_op_system_version` to set `CMAKE_SYSTEM_VERSION`